internal/provider/resources/resource_team_roles.go

```diff
@@ -0,0 +1,329 @@
+type teamRolesResource struct {
+	iamClient      *iam.ClientWithResponses
+	organizationId string
+}
+
+func (r *teamRolesResource) Metadata(
+	ctx context.Context,
+	req resource.MetadataRequest,
+	resp *resource.MetadataResponse,
+) {
+	resp.TypeName = req.ProviderTypeName + "_team_roles"
+}
+
+func (r *teamRolesResource) Schema(
+	ctx context.Context,
+	req resource.SchemaRequest,
+	resp *resource.SchemaResponse,
+) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Team Roles resource",
+		Attributes:          schemas.ResourceTeamRolesSchemaAttributes(),
+	}
+}
+
+func (r *teamRolesResource) Configure(
+	ctx context.Context,
+	req resource.ConfigureRequest,
+	resp *resource.ConfigureResponse,
+) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	apiClients, ok := req.ProviderData.(models.ApiClientsModel)
+	if !ok {
+		utils.ResourceApiClientConfigureError(ctx, req, resp)
+		return
+	}
+
+	r.iamClient = apiClients.IamClient
+	r.organizationId = apiClients.OrganizationId
+}
+
+func (r *teamRolesResource) MutateRoles(
+	ctx context.Context,
+	data *models.TeamRoles,
+) diag.Diagnostics {
+	teamId := data.TeamId.ValueString()
+
+	// Then convert the models to the request types for the API
+	workspaceRoles, diags := RequestWorkspaceRoles(ctx, data.WorkspaceRoles)
+	if diags.HasError() {
+		return diags
+	}
+	deploymentRoles, diags := RequestDeploymentRoles(ctx, data.DeploymentRoles)
+	if diags.HasError() {
+		return diags
+	}
+
+	// create request
+	updateTeamRolesRequest := iam.UpdateTeamRolesJSONRequestBody{
+		DeploymentRoles:  &deploymentRoles,
+		OrganizationRole: iam.UpdateTeamRolesRequestOrganizationRole(data.OrganizationRole.ValueString()),
+		WorkspaceRoles:   &workspaceRoles,
+	}
+	teamRoles, err := r.iamClient.UpdateTeamRolesWithResponse(
+		ctx,
+		r.organizationId,
+		teamId,
+		updateTeamRolesRequest,
+	)
+	if err != nil {
+		tflog.Error(ctx, "failed to mutate team_roles", map[string]interface{}{"error": err})
+		diags.AddError(
+			"Client Error",
+			fmt.Sprintf("Unable to mutate team_roles, got error: %s", err),
+		)
+		return diags
+	}
+	_, diagnostic := clients.NormalizeAPIError(ctx, teamRoles.HTTPResponse, teamRoles.Body)
+	if diagnostic != nil {
+		diags.Append(diagnostic)
+		return diags
+	}
+
+	diags = data.ReadFromResponse(ctx, teamId, teamRoles.JSON200)
+	if diags.HasError() {
+		return diags
+	}
+
+	return nil
+}
+
+func (r *teamRolesResource) Create(
+	ctx context.Context,
+	req resource.CreateRequest,
+	resp *resource.CreateResponse,
+) {
+	var data models.TeamRoles
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	diags := r.MutateRoles(ctx, &data)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
```

Comment:
could consider adding tflog.Trace logging here?

```suggestion
tflog.Trace(ctx, fmt.Sprintf("created a team_roles resource for team '%v'", data.TeamId.ValueString()))
```